### ISSUE TITLE
user's extraGroups list does not create groups

### ISSUE THREAD

**Issue by AMDphreak** [NixOS/nixpkgs#55557]
## Issue description

Declared a user including extraGroups with:
```
users.users.myuser = {
  extraGroups = [ "wheel" "networkmanager" ];
};
```

This compiled, but did absolutely nothing. No groups were created. Is there an unspecified requirement missing in the manual? If so, why are groups not automatically populated from configuration.nix? It requires root access so it should be given that power.

**Comment by edolstra** [comment]
This is not supposed to create groups, just specify group membership. It should probably give an error message if you're referencing a non-existent group, though.

**Comment by AMDphreak** [comment]
> This is not supposed to create groups, just specify group membership. It should probably give an error message if you're referencing a non-existent group, though.

I agree, a non-existent reference should throw an error. NixOS's current implementation may not be programmed to create the groups, but it *should* create the groups, as this is consistent with the declarative configuration model that NixOS is built around.
